### DISCUSSION TITLE
2.0.0-rc.1 / 2024-10-19

### DISCUSSION THREAD

**Discussion by galexrt** [galexrt/dellhw_exporter#138] [Announcements]
* [CHORE] add a basic nix flake to make development easier for me :-)
* [CHORE] replace logrus with zap logger
* [CHORE] update dependencies
* [CHORE] update golang version to 1.23
* [FEATURE] add `--collectors-additional` flag to allow enabling a disabled exporter on top of the `--collectors-enabled` flag
* [FEATURE] Helm chart: add `additionalEnv` list and update chart documentation
* [FEATURE] rewrite the omreport parser logic to be more flexible and consistent across `omreport` commands and versions (fixes #115 and other parsing related issues)
* [FIX] fix docs page not working (404 errors when opening any page)
* [FIX] updated documentation for new changes and added a FAQ page

Thanks for any discussions, feature requests, issues, and support for the project!

<hr /><em>This discussion was created from the release <a href='https://github.com/galexrt/dellhw_exporter/releases/tag/v2.0.0-rc.1'>2.0.0-rc.1 / 2024-10-19</a>.</em>

***

**Should you encounter any issues with the release candidate, please report them via the GitHub issues function, thanks!**

**Comment by sudhanvac** [comment]
Hi team,
When can we expect dellhw-exporter stable version? Please let us know the release plan.

Thank you

**Reply by galexrt** [reply]
@sudhanvac There haven't been any issues opened so I'm going to assume the release candidates were stable. I'm creating the v2.0.0 release at the time of writing this message.

Edit: here's the [release v2.0.0 page](https://github.com/galexrt/dellhw_exporter/releases/tag/v2.0.0).

**Reply by sudhanvac** [reply]
@galexrt , Thank you.
We were seeing few error logs from 2.0.0-rc.5, these logs are occasionally seen.

`failed","collector":"memory","duration":"9.195189361s","error":"failed to execute command (\"/opt/dell/srvadmin/bin/omreport [chassis memory -fmt ssv]\"). exit status 139"}
{"time":"2026-03-17T20:04:20.036286587Z","level":"ERROR","msg":"collector failed","collector":"storage_pdisk","duration":"9.79920938s","error":"failed to execute command (\"/opt/dell/srvadmin/bin/omreport [storage controller -fmt ssv]\"). exit status 139"}
{"time":"2026-03-17T20:04:20.099032193Z","level":"ERROR","msg":"collector failed","collector":"processors","duration":"9.864782679s","error":"failed to execute command (\"/opt/dell/srvadmin/bin/omreport [chassis processors -fmt ssv]\"). exit status 139"}
{"time":"2026-03-17T22:04:18.34779882Z","level":"ERROR","msg":"collector failed","collector":"fans","duration":"8.111012248s","error":"failed to execute command (\"/opt/dell/srvadmin/bin/omreport [chassis fans -fmt ssv]\"). exit status 139"}
{"time":"2026-03-17T22:04:18.359020524Z","level":"ERROR","msg":"collector failed","collector":"chassis","duration":"8.124411791s","error":"failed to execute command (\"/opt/dell/srvadmin/bin/omreport [chassis -fmt ssv]\"). exit status 139"}
{"time":"2026-03-17T22:04:19.562749012Z","level":"ERROR","msg":"collector failed","collector":"storage_pdisk","duration":"9.327135613s","error":"failed to execute command (\"/opt/dell/srvadmin/bin/omreport [storage controller -fmt ssv]\"). exit status 139"}`

Is this addressed in the stable version or this is something new?
Let me know.

Thanks.

**Reply by galexrt** [reply]
There haven't been reports of any such errors occurring before/during the release candidates.

If the errors only appear occasionally, as you wrote,  it is probably issues within `omreport`. The exit code `139` points to being a SIGSEGV (segfault).
You can, e.g., check if upgrading the server firmware helps, and if you are running dellhw_exporter as a binary directly on the server (not as a docker container) try updating `omreport` binary/packages.

**Reply by sudhanvac** [reply]
ok thanks @galexrt , we are using dellhw-exporter helm chart, so updating the omreport is not an option I guess. We will see if upgrading the server firmware can be done.
